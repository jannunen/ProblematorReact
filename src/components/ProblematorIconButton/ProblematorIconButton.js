import * as React  from 'react';
import {
  View,
  Text,
  StyleSheet,
  Picker,
  SectionList,
  TouchableOpacity
} from 'react-native'

import  FontAwesome  from 'react-native-vector-icons/FontAwesome5';
import  Icon  from 'react-native-vector-icons/Ionicons';
import { connect } from 'react-redux';

export class ProblematorIconButton extends React.Component {

    constructor(props){
        super(props)
      }


    onPressHandler = () => {
        this.props.onPress();
    }

    render() {
        return (
            <TouchableOpacity onPress={this.onPressHandler}>
                <FontAwesome style={[styles.iconStyle, this.props.style ]} {...this.props} />
            </TouchableOpacity>
        )
    }

}


const styles = StyleSheet.create({
    iconStyle : {
        fontSize : 30,
        color : 'white',
        padding : 8,

    }
});


const mapStateToProps = (state) => {
    return {
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(ProblematorIconButton);