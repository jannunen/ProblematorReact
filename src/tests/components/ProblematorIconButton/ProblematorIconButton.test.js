import 'react-native';
import React from 'react';
    jest.mock('TouchableOpacity', () => 'TouchableOpacity')
//import { TouchableOpacity } from 'react-native';
import sinon from 'sinon'
import configureMockStore from 'redux-mock-store'
import { shallow , mount} from 'enzyme';
import  Icon  from 'react-native-vector-icons/Ionicons';
import renderer from 'react-test-renderer';
import { Navigation} from 'react-native-navigation';
import thunk from 'redux-thunk'
import ProblematorIconButton from '../../../components/ProblematorIconButton/ProblematorIconButton';

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)


let  history, wrapper;

describe('ProblematorIconButton component tests',() => {

it('should correctly render ProblematorIconButton component', () => {
  const store = mockStore({ loading: false, error :false })
  const tree = shallow(<ProblematorIconButton name="ios-build" />);
  expect(tree).toMatchSnapshot();
});

it('should handle onPress',() => {

    const onPressEvent = jest.fn();
    onPressEvent.mockReturnValue('Link on press invoked');
    const wrapper = shallow(<ProblematorIconButton onPress={ onPressEvent } text='CustomLink Component'/>);
    wrapper.find('TouchableOpacity').props().onPress();
    expect( onPressEvent ).toHaveBeenCalled();

});


})