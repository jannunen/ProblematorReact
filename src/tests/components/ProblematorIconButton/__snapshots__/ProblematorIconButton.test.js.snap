// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProblematorIconButton component tests should correctly render ProblematorIconButton component 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
